docs/topics/GATEWAY.md

```diff
@@ -201,9 +203,29 @@ Used to request offline members for a guild. When initially connecting, the gate
 }
 ```
 
+### Gateway Hello
+
+Used to inform the client of the heartbeat value.
+
+###### Gateway Hello Structure
+
+| Field | Type | Description |
+|-------|------|-------------|
+| heartbeat_interval | integer | the interval (in seconds) the client should heartbeat with |
+| _trace | array of strings | used for debugging, array of servers connected to |
+
+###### Gateway Hello Example
+
+```json
+{
+	"heartbeat_interval": 45,
+	"_trace": ["discord-gateway-prd-1-99"]
+}
+```
+
 ## Connecting
 
-The first step to establishing a gateway connection is to request a gateway URL through the [Get Gateway](#DOCS_GATEWAY/get-gateway) API endpoint (if the client does not already have one cached). Using the "url" field from the response you can then create a new secure websocket connection that will be used for the duration of your gateway session. Once connected you must send an OP 2 [Identify](#DOCS_GATEWAY/gateway-identify-payload) or OP 6. If your token is correct, the gateway will respond with a [Ready](#DOCS_GATEWAY/ready-event) payload. After the ready payload, your client needs to start sending OP 1 [heartbeat](#DOCS_GATEWAY/gateway-heartbeat-payload) payloads every `heartbeat_interval` (which is sent in the ready payload) seconds.
+The first step to establishing a gateway connection is to request a gateway URL through the [Get Gateway](#DOCS_GATEWAY/get-gateway) API endpoint (if the client does not already have one cached). Using the "url" field from the response you can then create a new secure websocket connection that will be used for the duration of your gateway session. Once connected, the client will immediately receive an OP 10 [Hello](#DOCS_GATEWAY/gateway-hello) payload with the connection heartbeat interval. At this point the client should start sending OP 1 [heartbeat](#DOCS_GATEWAY/gateway-heartbeat) payloads every `heartbeat_interval` seconds. Next, the client sends an OP 2 [Identify](#DOCS_GATEWAY/gateway-identify) or OP 6 [Resume](#DOCS_GATEWAY/gateway-resume) payload. If your token is correct, the gateway will respond with a [Ready](#DOCS_GATEWAY/ready-event) payload.
```

Comment:
Worth noting during load or outage recovery, the gateway should always eventually respond. No client side logic for timing out a connection is needed as long as the gateway continues to heartbeat/ack.